### PR TITLE
chore/ci: add travis CI script

### PR DESCRIPTION
**Pull request by povilasb**
Appveyor is not used since travis integrated Windows support. Let's see how that works out for us :)

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+  - osx
+  - windows
+language: rust
+rust:
+  - 1.29.2
+sudo: false
+cache:
+  cargo: true
+before_script:
+  - cargo install cargo-prune --force
```
Comment (ustulation) [inline .travis.yml]: These are just going to slow things down. Can you pls use something [like this](https://github.com/ustulation/p2p/blob/example/.travis.yml) instead ?

File: .travis.yml [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+cache:
+  cargo: true
+before_script:
+  - if ! cargo prune --version | grep -q "0.1.8"; then
+      cargo install cargo-prune --vers="0.1.8" --force;
+    fi
+  - if ! rustfmt --version | grep -q "0.99.1-stable"; then
+      rustup component add rustfmt-preview;
+    fi
+  - if ! cargo clippy --version | grep -q "0.0.212"; then
```
Comment (ustulation) [inline .travis.yml]: This clippy version works with 1.29.0. Does it also work with 1.29.2 ? Just trying to make sure.

File: .travis.yml [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+      rustup component add clippy-preview;
+    fi
+script:
+  - cargo fmt -- --check &&
+    cargo test --verbose --release &&
+    cargo clippy --verbose --release &&
+    cargo clippy --verbose --release --profile=test
```
Comment (ustulation) [inline .travis.yml]: Again, doing it for all platforms is not going to buy much but rather slow down the build. Can we use the other link i gave you in previous review to do it only for one platform ? Also placing it above the actual binary formation (cargo test) might make more sense as we want to catch the early in the CI and bail out if possible.

File: .travis.yml (hunk 2 above)
Comment (povilasb) [reply .travis.yml]: It should, that's my setup:
```
➜  ~ rustc --version
rustc 1.29.2 (17a9dc751 2018-10-05)
➜  ~ rustfmt --version
rustfmt 0.99.1-stable (da17b68 2018-08-04)
➜  ~ cargo clippy --version
0.0.212
```

File: .travis.yml (hunk 3 above)
Comment (povilasb) [reply .travis.yml]: Not necessarily, windows/mac os/linux code might differ and all variations might have code style issues reported by clippy.

Comment (ustulation) [reply .travis.yml]: We don't have much platform dependent code and it doesn't buy us much to delay the CI. Besides we don't do it in any of our crates (no clippy on anything but Linux). If for the same code clippy produces diff results on different OSes then it's their problem to fix. So far i don't think we have had issues to make us implement clippy checks on all platforms.

**Pull request merged**